h4l/selection/lepton.py

```diff
@@ -0,0 +1,126 @@
+                                        (fSCeta>=1.479                & BDT > 0.79349796445)))
+
+            else: # UL WP (miniAOD branch Run2_CutBased_UL)
+                # print("This is not preUL!")
+                # from IPython import embed; embed()
+                return ((pt<=10.) &     (((fSCeta<0.8)                   & (BDT > 0.9128577458)) | \
+                                        ((fSCeta>=0.8) & (fSCeta<1.479) & (BDT > 0.9056792368)) | \
+                                        ((fSCeta>=1.479)                & (BDT > 0.9439440575)))) \
+                        | ((pt>10.) &  (((fSCeta<0.8)                   & (BDT > 0.1559788054)) | \
+                                        ((fSCeta>=0.8) & (fSCeta<1.479) & (BDT > 0.0273863727)) | \
+                                        ((fSCeta>=1.479)                & (BDT > -0.5532483665))))
+        return ak.ones_like(events.Electron.pt)
+    default_mask = (
+        (pt > min_pt) &
+        (abs(events.Electron.eta) < 2.5) & 
+        (events.Electron.dxy < 0.5) &
+        (events.Electron.dz < 1.0) & 
+        (abs(events.Electron.sip3d) < 4) & 
+        return_cuts()
+    )
+    sorted_idx = ak.argsort(events.Electron.pt, axis=1, ascending=False)
+    # filter unselected electron indices
+    selected_electron_idx = sorted_idx[default_mask[sorted_idx]]
+    
+
+
+    return events, SelectionResult(
+        objects = {
+            "Electron": {
+                "Electron": selected_electron_idx
+            }
+        }
+    )
+
+@selector(
+    uses=({
+        f"Muon.{var}" for var in [
+            # four momenta information
+            "pt", "eta", "phi", "mass", 
+            # quality criteria
+            "isGlobal", "isStandalone", "isTracker", "nStations", "nTrackerLayers",
+            # impact parameters
+            "dxy", "dz", "sip3d",
+            # IDs
+            "tightId", "mvaId", "highPtId",
+            #isolation
+            "pfRelIso03_all",
+        ]}
+    ),
+    exposed=False,
+)
+def muon_selector(self: Selector, events: ak.Array, **kwargs) -> tuple[ak.Array, SelectionResult]:
+    min_pt = 15
+    max_eta = 2.4
+    selected_muon_mask = (
+        # Global or Tracker Muon
+        (events.Muon.isGlobal | events.Muon.isTracker) &
+        # Discard Standalone Muon tracks if reconstructed in muon system only
+        (~events.Muon.isStandalone | (events.Muon.nTrackerLayers >0)) &
+        # WIP: Discard muons with muonBestTrackType==2 even if they are global or tracker muons
+        # --> muonBestTrackType not available?
+        (events.Muon.pt > min_pt) & 
+        (abs(events.Muon.eta) < max_eta) &
+        (events.Muon.dxy < 0.5) & (events.Muon.dz < 1.0) &
+        (abs(events.Muon.sip3d) < 4) &
+        # PF muon ID if pT < 200 GeV, PF muon ID or High-pT muon ID if pT > 200 GeV
+        (((events.Muon.pt > 200) & events.Muon.highPtId == 2) | (events.Muon.tightId)) &
```

Comment:
Again, for completeness, this is how this selection step is done in the main analysis:

```
mu.isPFcand or (mu.highPtId>0 and mu.pt>200.)
```